flink-ml-lib/src/main/java/org/apache/flink/ml/clustering/agglomerativeclustering/AgglomerativeClusteringParams.java

```diff
@@ -0,0 +1,107 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.clustering.agglomerativeclustering;
+
+import org.apache.flink.ml.common.param.HasDistanceMeasure;
+import org.apache.flink.ml.common.param.HasFeaturesCol;
+import org.apache.flink.ml.common.param.HasPredictionCol;
+import org.apache.flink.ml.param.BooleanParam;
+import org.apache.flink.ml.param.DoubleParam;
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.StringParam;
+
+/**
+ * Params of {@link AgglomerativeClustering}.
+ *
+ * @param <T> The class type of this instance.
+ */
+public interface AgglomerativeClusteringParams<T>
+        extends HasDistanceMeasure<T>, HasFeaturesCol<T>, HasPredictionCol<T> {
+    Param<Integer> K = new IntParam("k", "The max number of clusters to create.", 2);
+
+    Param<Double> DISTANCE_THRESHOLD =
+            new DoubleParam(
+                    "distanceThreshold",
+                    "Threshold to decide whether two clusters should be merged.",
+                    null);
+
+    String LINKAGE_WARD = "ward";
+    String LINKAGE_COMPLETE = "complete";
+    String LINKAGE_SINGLE = "single";
+    String LINKAGE_AVERAGE = "average";
+    /**
+     * Supported options to compute the distance between two clusters.
+     *
+     * <ul>
+     *   <li>ward: difference between the sum of the variance of the two clusters and the merged
+     *       one.
+     *   <li>complete: the maximum distance between all observations of the two clusters.
+     *   <li>single: the minimum distance between all observations of the two cluster.
+     *   <li>average: the average of the distance of all observations of the two cluster.
+     * </ul>
+     */
+    Param<String> LINKAGE =
+            new StringParam(
+                    "linkage",
+                    "Criterion for computing distance between two clusters.",
+                    LINKAGE_WARD,
+                    ParamValidators.inArray(
+                            LINKAGE_WARD, LINKAGE_COMPLETE, LINKAGE_AVERAGE, LINKAGE_SINGLE));
+
+    Param<Boolean> COMPUTE_FULL_TREE =
+            new BooleanParam(
+                    "computeFullTree",
+                    "Whether computes the full tree after convergence.",
+                    false,
```

Comment:
The default behavior here is different from that in sklearn. Do you think we should also support the `auto` option here?

Reply:
With some offline discussion, we aggree that not supporting `auto` is ok for users.